lib/tokenize.cpp

```diff
@@ -1874,8 +1877,28 @@ namespace {
 
         return false;
     }
+
+    std::string memberFunctionScope(const Token *tok)
+    {
+        std::string qualification;
+        const Token *qualTok = tok->strAt(-2) == "~" ? tok->tokAt(-4) : tok->tokAt(-3);
+        while (Token::Match(qualTok, "%type% ::")) {
+            if (!qualification.empty())
+                qualification = " :: " + qualification;
+            qualification = qualTok->str() + qualification;
+            qualTok = qualTok->tokAt(-2);
+        }
+        return qualification;
+    }
 } // namespace
 
+bool Tokenizer::isMemberFunction(const Token *openParen) const
```

Comment:
Wouldn't this match a function definition in a namespace as well? Could this cause any trouble?

Reply:
Yes it would find a function in a namespace and it should. I guess the name could be changed to something like  isFunctionThatIsDeclaredInAnotherScope to be technically correct but a member function is the most common case.